api-rights-bridge/src/test/java/org/xwiki/contrib/rights/internal/DefaultRightsWriterTest.java

```diff
@@ -409,6 +425,35 @@ void testSaveRuleWithNullRights() throws XWikiException
         assertEquals(1, rightsObj.getIntValue(ALLOW_PROPERTY));
     }
 
+    @Test
+    void testSaveRuleWithEmptyRightsList() throws XWikiException
+    {
+        DocumentReference documentReference = new DocumentReference("xwiki", "S", "P");
+        // prepare rules to put on the document, with empty rights set
+        WritableSecurityRule rule = new WritableSecurityRuleImpl(
+            Collections.singletonList(new DocumentReference("xwiki", "XWiki", "XWikiAdminGroup")),
+            Collections.emptyList(), new RightSet(Right.VIEW), RuleState.ALLOW);
+
+        // call the function under test
+        this.rightsWriter.saveRules(Arrays.asList(rule), documentReference);
+
+        // get the document that was just modified and assert on it
+        XWikiDocument resultDoc =
+            this.oldcore.getSpyXWiki().getDocument(documentReference, this.oldcore.getXWikiContext());
+        // check that there is a single object set
+        assertEquals(1, resultDoc.getXObjects(DefaultRightsWriter.XWIKI_RIGHTS_CLASS).size());
+
+        // Clean all rights objects
```

Comment:
this comment should be "Remove all access rules": from the point of view of the rights API, it doesn't care how access rules are stored (objects or other), so when it removes it removes the abstract object that is a security rule